make test fails with `make: *** [Makefile:73: test] Error 1`

Running `make test` on my environment results in:
```bash
15:37 $ make test
cd tools/setup-envtest && go test ./...
?   	sigs.k8s.io/controller-runtime/tools/setup-envtest	[no test files]
?   	sigs.k8s.io/controller-runtime/tools/setup-envtest/remote	[no test files]
ok  	sigs.k8s.io/controller-runtime/tools/setup-envtest/env	(cached)
ok  	sigs.k8s.io/controller-runtime/tools/setup-envtest/store	(cached)
ok  	sigs.k8s.io/controller-runtime/tools/setup-envtest/versions	(cached)
ok  	sigs.k8s.io/controller-runtime/tools/setup-envtest/workflows	(cached)
TRACE=1 ./hack/check-everything.sh
++ go mod edit -json
++ MODULES_ENABLED=0
++ MOD_OPT=
++ MODULES_OPT=
++ [[ -n '' ]]
++ NO_COLOR=
++ '[' -z '' ']'
++ header=''
++ reset=''
+ tmp_root=/tmp
+ kb_root_dir=/tmp/kubebuilder
++ make go-version
+ export 'GOTOOLCHAIN=gomake[1]: Entering directory '\''/home/valdar/projects/communityProjects/controller-runtime'\''
1.23.0
make[1]: Leaving directory '\''/home/valdar/projects/communityProjects/controller-runtime'\'''
+ GOTOOLCHAIN='gomake[1]: Entering directory '\''/home/valdar/projects/communityProjects/controller-runtime'\''
1.23.0
make[1]: Leaving directory '\''/home/valdar/projects/communityProjects/controller-runtime'\'''
+ ./hack/verify.sh
++ go mod edit -json
make: *** [Makefile:73: test] Error 1

```
I recon that the error is caused by the `GOTOOLCHAIN` being wrongly evaluated, in fact if I add the `--silent` option to these make invocations: 

- https://github.com/kubernetes-sigs/controller-runtime/blob/6e9496d8cdbe5075a3640de31650ce584cc39329/hack/check-everything.sh#L27
- https://github.com/kubernetes-sigs/controller-runtime/blob/6e9496d8cdbe5075a3640de31650ce584cc39329/hack/apidiff.sh#L26

everything seems to work as expected.

--- a/hack/apidiff.sh
+++ b/hack/apidiff.sh
@@ -23,7 +23,7 @@ source $(dirname ${BASH_SOURCE})/common.sh
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 cd "${REPO_ROOT}"
 
-export GOTOOLCHAIN="go$(make go-version)"
+export GOTOOLCHAIN="go$(make --silent go-version)"
 
 header_text "verifying api diff"
 echo "*** Running go-apidiff ***"

--- a/hack/check-everything.sh
+++ b/hack/check-everything.sh
@@ -24,7 +24,7 @@ source ${hack_dir}/common.sh
 tmp_root=/tmp
 kb_root_dir=$tmp_root/kubebuilder
 
-export GOTOOLCHAIN="go$(make go-version)"
+export GOTOOLCHAIN="go$(make --silent go-version)"
 
 # Run verification scripts.
 ${hack_dir}/verify.sh

